zebra-chain/src/parameters/network.rs

```diff
@@ -51,52 +53,112 @@ mod tests;
 /// after the grace period.
 const ZIP_212_GRACE_PERIOD_DURATION: HeightDiff = 32_256;
 
-/// An enum describing the possible network choices.
+/// Network consensus parameters for test networks such as Regtest and the default Testnet.
 #[derive(Clone, Debug, Default, Eq, PartialEq, Hash, Serialize, Deserialize)]
 #[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
-pub enum Network {
+pub struct NetworkParameters {}
+
+impl NetworkParameters {
+    /// Returns true if the instance of [`NetworkParameters`] represents the default public Testnet.
+    pub fn is_default_testnet(&self) -> bool {
+        self == &Self::default()
```

Comment:
###### Optional Suggestion

```suggestion
        // TODO: Change this comparison once `Self` gets populated, since
        // it currently uses `default`, but the default `NetworkKind` is Mainnet.
        self == &Self::default()
```

Reply:
The default network is also Mainnet, we're only using `NetworkParameters` for Testnet, do we want to rename `NetworkParameters` to `TestnetParameters` or move it to a sub-module to make it `network::testnet::Parameters`?

Reply:
> move it to a sub-module to make it `network::testnet::Parameters`?

This sounds great to me.

Reply:
Sorry for the confusing comment, I forgot the params are only related to Testnet.

Reply:
If we can't go with `testnet::Parameters` for any reason, it think it would be good to at least rename the struct to `TestnetParameters`.

Reply:
Moved to a `testnet` module in https://github.com/ZcashFoundation/zebra/pull/8368/commits/365b58ea3f874d09f54de898f3c1dc8455f2bb77. This helps move the `default_testnet_params()` method in the following PR to a `Default` trait impl too, good idea!